src/palette/internal/palettelayout.cpp

```diff
@@ -857,8 +1035,383 @@ void PaletteLayout::layout(TimeSig* item, const Context& ctx)
     item->setDrawArgs(drawArgs);
 }
 
-//! TODO
-void PaletteLayout::layout(Volta*, const Context&)
+void PaletteLayout::layout(Vibrato* item, const Context& ctx)
+{
+    layoutLine(static_cast<SLine*>(item), ctx);
+}
+
+void PaletteLayout::layout(VibratoSegment* item, const Context&)
+{
+    if (item->spanner()->placeBelow()) {
+        item->setPosY(item->staff() ? item->staff()->height() : 0.0);
+    }
+
+    switch (item->vibrato()->vibratoType()) {
+    case VibratoType::GUITAR_VIBRATO:
+        item->symbolLine(SymId::guitarVibratoStroke, SymId::guitarVibratoStroke);
+        break;
+    case VibratoType::GUITAR_VIBRATO_WIDE:
+        item->symbolLine(SymId::guitarWideVibratoStroke, SymId::guitarWideVibratoStroke);
+        break;
+    case VibratoType::VIBRATO_SAWTOOTH:
+        item->symbolLine(SymId::wiggleSawtooth, SymId::wiggleSawtooth);
+        break;
+    case VibratoType::VIBRATO_SAWTOOTH_WIDE:
+        item->symbolLine(SymId::wiggleSawtoothWide, SymId::wiggleSawtoothWide);
+        break;
+    }
+
+    item->setOffset(PointF());
+}
+
+void PaletteLayout::layout(Volta* item, const Context& ctx)
+{
+    layoutLine(item, ctx);
+}
+
+void PaletteLayout::layout(VoltaSegment* item, const Context& ctx)
+{
+    layoutTextLineBaseSegment(item, ctx);
+    item->setOffset(PointF());
+    item->text()->setOffset(PointF(10.0, 54.0)); //! TODO
+}
+
+void PaletteLayout::layout(Text* item, const Context& ctx)
+{
+    layoutTextBase(static_cast<TextBase*>(item), ctx);
+}
+
+void PaletteLayout::layoutTextBase(TextBase* item, const Context& ctx)
 {
-    // layoutLine(item, ctx);
+    item->setPos(PointF());
+    item->setOffset(PointF());
+
+    if (item->placeBelow()) {
+        item->setPosY(0.0);
+    }
+
+    layout1TextBase(item, ctx);
+}
+
+void PaletteLayout::layout1TextBase(TextBase* item, const Context&)
+{
+    if (item->isBlockNotCreated()) {
+        item->createBlocks();
+    }
+    if (item->blocksRef().empty()) {
```

Comment:
It seems that createBlocks() already does this

Reply:
I am porting previously written code, without detailed investigation. If the author wrote this, then there are situations when this is possible - for example, the text is empty.